spring-xsuaa/src/test/java/com/sap/cloud/security/xsuaa/token/flows/RefreshTokenFlowTests.java

```diff
@@ -0,0 +1,223 @@
+					TestConstants.tokenEndpointUri, TestConstants.authorizeEndpointUri,
+					TestConstants.keySetEndpointUri);
+			tokenFlow.client(clientId)
+					.secret(clientSecret)
+					.execute();
+		}).isInstanceOf(TokenFlowException.class).hasMessageContaining("Refresh token not set");
+
+		assertThatThrownBy(() -> {
+			RefreshTokenFlow tokenFlow = new RefreshTokenFlow(restTemplate, tokenDecoder,
+					TestConstants.tokenEndpointUri, TestConstants.authorizeEndpointUri,
+					TestConstants.keySetEndpointUri);
+			tokenFlow.refreshToken("dummy")
+					.execute();
+		}).isInstanceOf(TokenFlowException.class).hasMessageContaining("Refresh token flow request is not valid");
+
+		assertThatThrownBy(() -> {
+			RefreshTokenFlow tokenFlow = new RefreshTokenFlow(restTemplate, tokenDecoder,
+					TestConstants.tokenEndpointUri, TestConstants.authorizeEndpointUri,
+					TestConstants.keySetEndpointUri);
+			tokenFlow.client(null)
+					.secret(clientSecret)
+					.execute();
+		}).isInstanceOf(IllegalArgumentException.class).hasMessageContaining("client ID");
+
+		assertThatThrownBy(() -> {
+			RefreshTokenFlow tokenFlow = new RefreshTokenFlow(restTemplate, tokenDecoder,
+					TestConstants.tokenEndpointUri, TestConstants.authorizeEndpointUri,
+					TestConstants.keySetEndpointUri);
+			tokenFlow.client(clientId)
+					.secret(null)
+					.execute();
+		}).isInstanceOf(IllegalArgumentException.class).hasMessageContaining("client secret");
+	}
+
+	@Test
+	public void test_execute() throws TokenFlowException {
+
+		HttpEntity<Void> expectedRequest = buildExpectedRequest(clientId, clientSecret);
+
+		URI expectedRequestURI = UriComponentsBuilder.fromUri(TestConstants.tokenEndpointUri)
+				.queryParam("grant_type", "refresh_token")
+				.queryParam("refresh_token", refreshToken)
+				.build().toUri();
+
+		RestTemplateMock restTemplateMock = new RestTemplateMock(expectedRequestURI, expectedRequest, Map.class,
+				mockJwt.getTokenValue(), HttpStatus.OK);
+
+		RefreshTokenFlow tokenFlow = new RefreshTokenFlow(restTemplateMock, tokenDecoderMock,
+				TestConstants.tokenEndpointUri, TestConstants.authorizeEndpointUri, TestConstants.keySetEndpointUri);
+		tokenFlow.refreshToken(refreshToken)
+				.client(clientId)
+				.secret(clientSecret)
+				.execute();
+
+		restTemplateMock.validateCallstate();
+		tokenDecoderMock.validateCallstate();
+	}
+
+	@Test
+	public void test_execute_throwsIfHttpStatusUnauthorized() throws TokenFlowException {
+
+		HttpEntity<Void> expectedRequest = buildExpectedRequest(clientId, clientSecret);
+
+		URI expectedRequestURI = UriComponentsBuilder.fromUri(TestConstants.tokenEndpointUri)
+				.queryParam("grant_type", "refresh_token")
+				.queryParam("refresh_token", refreshToken)
+				.build().toUri();
+
+		RestTemplateMock restTemplateMock = new RestTemplateMock(expectedRequestURI, expectedRequest, Map.class,
+				mockJwt.getTokenValue(), HttpStatus.UNAUTHORIZED);
```

Comment:
need to fix this test as an unauthorized is represented by a HttpClientErrorException